codalab/server/authenticated_plugin.py

```diff
@@ -59,16 +94,44 @@ class AuthenticatedPlugin(object):
       - Requests from the CLI.
     """
 
-    api = 2
+    def apply(self, callback, route):
+        def wrapper(*args, **kwargs):
+            self.check_user_authenticated()
+            return callback(*args, **kwargs)
+
+        return wrapper
+
+
+class AuthenticatedProtectedPlugin(AuthenticationPlugin):
+    """
+    Fails the request if the user is not authenticated. It doesn't matter if the instance is protected or not.
```

Comment:
I think this should clarify the comment more:

```suggestion
    Fails the request if the user is not authenticated. If the instance is protected, fails the request if the user has not been granted access to the instance.
```